docs/skills/README.md

```diff
@@ -0,0 +1,237 @@
+# Skills System
+
+The skills system extends gptme's lesson system to support bundled tools, scripts, and workflows inspired by Claude's Skills system and Cursor's rules system.
+
+## Overview
+
+**Skills** are enhanced lessons that bundle:
+- Instructional content (like lessons)
+- Executable scripts and utilities
+- Dependencies and setup requirements
+- Hook points for automation
+
+Skills complement lessons by providing **executable components** alongside guidance.
+
+## Skill vs. Lesson
+
+| Feature | Lesson | Skill |
+|---------|--------|-------|
+| Purpose | Guidance and patterns | Executable workflows |
+| Content | Instructions, examples | Instructions + scripts |
+| Scripts | None | Bundled helper scripts |
+| Dependencies | None | Explicit package requirements |
+| Hooks | None | Pre/post execution hooks |
+
+**When to use**:
+- **Lesson**: Teaching patterns, best practices, tool usage
+- **Skill**: Providing reusable scripts, automated workflows, integrated tooling
+
+## Skill Format
+
+Skills use the same YAML frontmatter as lessons with additional fields:
+
+```yaml
+---
+type: skill  # Distinguishes from lessons
+status: active
+match:
+  keywords: [python, data]
+  tools: [ipython]
+scripts:
+  - helper_script.py
+  - utilities.py
+dependencies:
+  - numpy
+  - pandas
+hooks:
+  - pre_execute
+  - post_execute
+---
+
+# Skill Title
+
+Skill description and usage instructions...
+```
+
+### Metadata Fields
+
+#### Required
+- `type`: Must be "skill" (defaults to "lesson")
+- `status`: active, automated, deprecated, or archived
+- `match`: Trigger keywords and tools
+
+#### Optional
+- `scripts`: List of bundled script files (relative to skill directory)
+- `dependencies`: Required Python packages
+- `hooks`: Execution hooks (pre_execute, post_execute)
+
+## Directory Structure
+
+Skills are organized parallel to lessons:
+
+gptme/
+├── lessons/           # Instructional patterns
+│   ├── tools/
+│   ├── patterns/
+│   └── workflows/
+└── skills/            # Executable workflows
+    ├── examples/
+    │   ├── python-repl-skill.md
+    │   └── python_helpers.py
+    └── custom/
+
+## Creating Skills
+
+### 1. Design the Skill
+
+Identify:
+- What workflow or automation does it provide?
+- What scripts/utilities are needed?
+- What dependencies are required?
+- Where should hooks run?
+
+### 2. Create Skill File
+
+Create `skill-name.md` in appropriate directory:
+
+```yaml
+---
+type: skill
+status: active
+match:
+  keywords: [relevant, keywords]
+  tools: [tool1, tool2]
+scripts:
+  - script1.py
+  - script2.py
+dependencies:
+  - package1
+  - package2
+hooks:
+  - pre_execute
+---
+
+# Skill Title
+
+## Overview
+Brief description of what the skill does.
+
+## Bundled Scripts
+Describe each script and its purpose.
+
+## Usage Patterns
+Show common usage examples.
+
+## Hooks
+Explain what each hook does.
+
+## Dependencies
+List and explain required packages.
+```
```

Comment:
Should be SKILL.md in an appropriate directory (`lessons/skills/skill-name/SKILL.md`)